src/utils/retry_utils/src/retry_utils.c

```diff
@@ -58,7 +58,7 @@ time_t ADUC_Retry_Delay_Calculator(
     {
         delay = (double)maxDelaySecs;
     }
-    time_t retryTimestampSec =
-        (time_t)(GetTimeSinceEpochInSeconds() + (time_t)additionalDelaySecs + (time_t)(delay * (1 + jitterPercent)));
+    time_t retryTimestampSec = (time_t)(
+        ADUC_GetTimeSinceEpochInSeconds() + (time_t)additionalDelaySecs + (time_t)(delay * (1 + jitterPercent)));
```

Comment:
I'm not comfortable with all the casting between time_t and double

Reply:
on some platforms, time_t can be other types such as float, long double, long long